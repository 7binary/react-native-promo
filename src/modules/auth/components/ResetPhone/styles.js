import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    paddingVertical: 10,
  },
  icon: {
    marginLeft: 10,
    marginTop: -6,
  },
});
