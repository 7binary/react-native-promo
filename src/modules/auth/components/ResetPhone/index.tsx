import React from 'react';
import { Text, View } from 'react-native';
import Ionicons from 'react-native-vector-icons/Ionicons';

import styles from './styles';
import { Settings } from '@pmk-team/common';

interface Props {
  phone: string | null | undefined,
  onReset: Function,
}

const ResetPhone: React.FC<Props> = ({ phone, onReset }) => {
  if (!phone) {
    return null;
  }
  return (
    <View style={styles.container}>
      <Text>{phone}</Text>
      <Ionicons
        style={styles.icon}
        name="ios-backspace"
        size={26}
        color={Settings.colors.formIcon}
        onPress={() => onReset()}
      />
    </View>
  );
};

export default ResetPhone;
