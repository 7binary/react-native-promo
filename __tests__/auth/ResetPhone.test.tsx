import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ResetPhone from 'modules/auth/components/ResetPhone';

it('renders correctly', () => {
  const tree = renderer
    .create(<ResetPhone phone={'+79299004050'} onReset={jest.fn()}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
