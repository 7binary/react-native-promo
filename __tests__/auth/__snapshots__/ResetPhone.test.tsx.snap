// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
      "margin": 15,
      "marginTop": 0,
      "paddingHorizontal": 20,
      "paddingTop": 20,
    }
  }
>
  <Text>
    +79299004050
  </Text>
  <Text
    allowFontScaling={false}
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": "#444",
          "fontSize": 26,
        },
        Object {
          "marginLeft": 10,
          "marginTop": -4,
        },
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ï…‘
  </Text>
</View>
`;
